### PR TITLE
Eng 1369 allow users to delete workflows and associated written objects in SDK

### PR DESCRIPTION
**Pull request by eunice-chan**
## Describe your changes and why you are making these changes
Add the ability to delete any objects written by the workflow when deleting the workflow in the SDK.

## Related issue number (if any)
ENG 1394
ENG-1369 (builds on above)
ENG-1325 (builds on above)
ENG-1431 (builds on above)

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [N/A] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [x] If this is a new feature, I have added unit tests and integration tests.
    - **Integration tests won't work because BE not set up. Added some extra data to be sent to the delete workflow endpoint and as a result the current endpoint fails. See the PR with the BE part for the integration tests**
- [N/A] I have run the integration tests locally and they are passing.
    - **Integration tests won't work because BE not set up. Added some extra data to be sent to the delete workflow endpoint and as a result the current endpoint fails. See the PR with the BE part for the integration tests**
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [N/A] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)
    - **Integration tests won't work because BE not set up. Added some extra data to be sent to the delete workflow endpoint and as a result the current endpoint fails. See the PR with the BE part for the integration tests**

### CODE REVIEW

File: sdk/aqueduct/aqueduct_client.py [hunk 1]
```diff
@@ -417,12 +418,21 @@ def trigger(
         flow_id = parse_user_supplied_id(flow_id)
         self._api_client.refresh_workflow(flow_id, serialized_params)
 
-    def delete_flow(self, flow_id: Union[str, uuid.UUID]) -> None:
+    def delete_flow(
+        self,
+        flow_id: Union[str, uuid.UUID],
+        writes_to_delete: Optional[DefaultDict[uuid.UUID, List[WrittenObject]]] = None,
```
Comment (likawind) [inline sdk/aqueduct/aqueduct_client.py:424]: I feel we could relate to 'save' which is a concept available for the rest of sdk. For example, `saved_objects_to_delete`. cc @kenxu95 @vsreekanti

File: sdk/aqueduct/aqueduct_client.py [hunk 2]
```diff
@@ -417,12 +418,21 @@ def trigger(
...
+        force: bool = False,
+    ) -> Dict[uuid.UUID, List[WritesDelete]]:
```
Comment (likawind) [inline sdk/aqueduct/aqueduct_client.py:426]: Should we return any objects here? I don't think if the user would actually programmatically use this returned value  esp. that it's also an `uuid->list` mapping. If it's not intended for user to use the returned value, we should return None here (and it's also generally natural that a `delete` returns `None`)

It should be sufficient that we print out any deletion failures for saved object. We may actually want to raise exceptions in these failures.

File: sdk/aqueduct/aqueduct_client.py [hunk 3]
```diff
@@ -431,11 +441,31 @@ def delete_flow(self, flow_id: Union[str, uuid.UUID]) -> None:
...
+                    failed_for_integration.append(obj)
+            if len(failed_for_integration) > 0:
+                failed_deletions[str(integration)] = [
+                    res.__dict__ for res in failed_for_integration
+                ]
+                counts += len(failed_for_integration)
+        if counts > 0:
+            print("Workflow-Written Objects' Deletion Failures")
+            print(f"{counts} Failures")
+            print(json.dumps(failed_deletions, sort_keys=False, indent=4))
```
Comment (likawind) [inline sdk/aqueduct/aqueduct_client.py:467]: this will print the UUID as keys. I feel we would also want to print the integration name instead.

**Review by likawind — CHANGES_REQUESTED** [review]
Overall looks good! Some comments around naming and user messaging

File: sdk/aqueduct/aqueduct_client.py (hunk 1 above)
Comment (kenxu95) [reply sdk/aqueduct/aqueduct_client.py:424]: yeah thats fine with me. I agree, 'save' exposes the user to the same terminology as our API.

File: sdk/aqueduct/aqueduct_client.py (hunk 2 above)
Comment (eunice-chan) [reply sdk/aqueduct/aqueduct_client.py]: Returning None now and raising an exception if any table failed to delete

**Review by likawind — APPROVED** [review]

**Pull request merged**